src/controllers/cart.controller.ts

```diff
@@ -0,0 +1,23 @@
+import { RequestWithUserId } from "src/types/types";
+import * as cartService from "../services/cart.service";
+import { Request, Response } from "express";
+
+export const addProductToCart = (req: Request, res: Response) => {

```

Comment:
can you set type `RequestWithUserId` for the `req` then it won't be needed below

Reply:
![image](https://github.com/user-attachments/assets/686a147c-d66e-4daf-ae04-bebda2f70d68)
I'm having an error when using type `RequestWithUserId` for the `req`.
I decided to create a helper function 'getUserId' and remove this 'RequestWithUserId' type.